custom/icds/messaging/indicators.py

```diff
@@ -224,3 +286,73 @@ def convert_to_date(date):
             messages.append(self.render_template(context, language_code=language_code))
 
         return messages
+
+
+class LSAggregatePerformanceIndicator(LSIndicator):
+    template = 'ls_aggregate_performance.txt'
+
+    @property
+    @memoized
+    def restore_user(self):
+        return OTARestoreCommCareUser(self.domain, self.user)
+
+    @property
+    @memoized
+    def report_configs(self):
+        report_ids = [
+            HOME_VISIT_REPORT_ID,
+            THR_REPORT_ID,
+            CHILDREN_WEIGHED_REPORT_ID,
+            DAYS_AWC_OPEN_REPORT_ID,
+        ]
+        app = get_app(self.domain, SUPERVISOR_APP_ID)
+        return {
+            report_config.report_id: report_config
+            for module in app.modules if isinstance(module, ReportModule)
+            for report_config in module.report_configs
+            if report_config.report_id in report_ids
+        }
+
+    def get_report_fixture(self, report_id):
+        return ReportFixturesProvider.report_config_to_fixture(
+            self.report_configs[report_id], self.restore_user
+        )
+
+    @property
+    def visits_fixture(self):
+        return self.get_report_fixture(HOME_VISIT_REPORT_ID)
+
+    @property
+    def thr_fixture(self):
+        return self.get_report_fixture(THR_REPORT_ID)
+
+    @property
+    def weighed_fixture(self):
+        return self.get_report_fixture(CHILDREN_WEIGHED_REPORT_ID)
+
+    @property
+    def days_open_fixture(self):
+        return self.get_report_fixture(DAYS_AWC_OPEN_REPORT_ID)
+
+    def get_value_from_fixture(self, fixture, attribute):
```

Comment:
can we add the same IndicatorError exception raising here as you did for the AWW one?